pytermgui/ansi_interface.py

```diff
@@ -282,24 +293,30 @@ def fill(self, color: int = 0, flush: bool = True) -> None:
         if flush:
             sys.stdout.flush()
 
+    @contextmanager
+    def NoEcho(self) -> None:
+        if _IS_NT:
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(-10), 0)
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(-11), 7)
+        else:
+            _ = self._OldStdinMode
+            _[3] = _[3] & ~(termios.ECHO | termios.ICANON)
+            termios.tcsetattr(sys.stdin, termios.TCSAFLUSH, _)
+        try:
+            yield
+        finally:
+            if _IS_NT:
+                kernel32.SetConsoleMode(kernel32.GetStdHandle(-10), self._OldStdinMode)
+                kernel32.SetConsoleMode(kernel32.GetStdHandle(-11), self._OldStdoutMode)
+            else:
+                termios.tcsetattr(sys.stdin, termios.TCSAFLUSH, self._OldStdinMode)
+
 
 terminal = _Terminal()
 
 # helpers
-def _tput(command: list[str]) -> None:
-    """Shorthand for tput calls"""
-
-    waited_commands = ["clear", "smcup", "cup"]
-
-    command.insert(0, "tput")
-    str_command = [str(c) for c in command]
-
-    if command[1] in waited_commands:
-        _run(str_command, check=True)
-        return
-
-    # A with statement here would result in `pass`, so is unnecessary.
-    _Popen(str_command)  # pylint: disable=consider-using-with
+# EDIT: removed `_tcup` function since we don't need it anymore.
+# I replaced it with ANSI sequences
```

Comment:
On some systems the ANSI sequences don't work. Those calls were needed.

Reply:
It's not system that defines ANSI support but terminal emulator itself. **tput** also uses escape sequences

Reply:
Look. I am aware of how these systems work; I managed to write this library somehow. My problem here is that I had a real compatibility issue with a device I was using to code on, where ANSI sequences did not work, but `tput` did. I am aware of the compatibility issues, but I still had that problem.

As I currently cannot test it (I do not have my old device at the moment) I'll let this pass.